cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -193,7 +185,25 @@ public ProgramScheduleStatus scheduleState(ProgramId program, SchedulableProgram
                                              String scheduleName)
     throws SchedulerException, NotFoundException {
     Scheduler scheduler = getSchedulerForSchedule(program, scheduleName);
-    return scheduler.scheduleState(program, programType, scheduleName);
+    ProgramScheduleStatus status;
+    try {
+      status = scheduler.scheduleState(program, programType, scheduleName);
+    } catch (NotFoundException e) {
```

Comment:
For other methods, since they are all called from methods in `CoreSchedulerService`, and those methods in `CoreSchedulerService` will check whether `CoreSchedulerService` has been fully initialized. This method can propagate ServiceNotAvailableException to `CoreSchedulerService#migrateSchedules`, so that `constraintCheckerService` and `notificationSubscriberService` won't be started until `migrateSchedules` is done